Rework implementation of the Properties admin facet in C#

bernardnormier opened the pull request:
This tiny PR mostly changes `INativePropertyAdmin`, which is now using an event.

pepone commented on csharp/src/Ice/PropertiesAdmin.cs:
```diff
@@ -7,33 +7,31 @@
 
-        public void
-        SetProperties(Dictionary<string, string> props, Ice.Current current)
+        public Dictionary<string, string> GetPropertiesForPrefix(string name, Current current) =>
```
I will use `prefix` for the argument here instead of `name`

bernardnormier commented:
I've just updated it further and replaced `INativePropertiesAdmin` by a partial interface `IPropertiesAdmin`. This way, no need for this "native" interface. I also added `partial` to the generation of interface proxies and skeletons.

Also synchronized the parameter names in PropertiesAdmin with those used in the Slice definition.

pepone commented on csharp/src/Ice/PropertiesAdmin.cs:
```diff
@@ -7,9 +7,9 @@
-    /// to the Slice interface PropertiesAdmin) and the local INativePropertiesAdmin interface.</summary>
-    public interface INativePropertiesAdmin
+    /// <summary>This partial interface extends the IPropertieAdmin interface generated from the Slice interface
+    /// PropertiesAdmin. All Properties admin facet implement this interface.</summary>
```
`All Properties admin facet implement this interface`
I don't understand this sentence, what do you mean?

bernardnormier replied:
I removed it as it's indeed unclear here.
